pkg/cache/context.go

```diff
@@ -601,25 +601,18 @@ func (ctx *Context) deletePriorityClass(obj interface{}) {
 }
 
 func (ctx *Context) triggerReloadConfig(index int, configMap *v1.ConfigMap) {
-	ctx.lock.Lock()
-	defer ctx.lock.Unlock()
-
-	conf := ctx.apiProvider.GetAPIs().GetConf()
-	if !conf.EnableConfigHotRefresh {
+	// hot reload is turned off do nothing
+	// hot reload can be turned off by an update: safety first access under lock to prevent data race
+	if !ctx.apiProvider.GetAPIs().GetConf().IsConfigReloadable() {
```

Comment:
This is unrelated to this PR, but should we replace this long call chain by`conf.GetSchedulerConf().IsConfigReloadable()`? The `SchedulerConf` is a global singleton, and it would be nice that we unify the usage of `ctx.apiProvider.GetAPIs().GetConf()` and `conf.GetSchedulerConf()`

Reply:
removed the call and use scheduler conf directly